### PR TITLE
initial updates to network documentation

### PR DESCRIPTION
**Pull request by njd42**
Mainly tidying and adding some additional context for the multiplexer

### CODE REVIEW

File: docs/network-spec/connection-manager.tex [hunk 1]
```diff
@@ -273,7 +356,212 @@ \section{Components}
...
+\item[Query number of connections] The server component needs to query the
+  connection manager to find the current number of connections. It uses this
+  information to decide if any new connections can be accepted or if we are
+  at the resource limits. Below the hard limits, the current number can be used
+  as part of rate limiting decisions.
+
+\item[Hand over a new connection] Once the server component has successfully
+  accepted a new connection, it needs to hand over responsibility for it to
+  the connection manager.
+  \todo[inline]{be clearer in the responsiblities and the conditions that should hold}
```
Comment (bolt12) [inline docs/network-spec/connection-manager.tex:450]: I don't think there's much to say here besides that only connections that are allowed by rate limiting make it through

File: docs/network-spec/connection-manager.tex [hunk 2]
```diff
@@ -273,7 +356,212 @@ \section{Components}
...
+handler.
+
+All the mini-protocols have the property that agency starts with the remote
+peer\footnote{Originally transaction submission protocol had agency start with the
+server side. A later protocol update reversed the initial agency so that they
+are now all consistent.}. This allows all of the server side protocols to be
+started in the mux `on-demand' mode. In the on-demand mode the protocol handler
+thread is not started until the first message arrives from the client.
+\todo[inline]{this lack of active processing locally is used as a witness of remote
+      peer idleness - where should we be explicit about that?}
```
Comment (bolt12) [inline docs/network-spec/connection-manager.tex:482]: I don't know what this is about

File: docs/network-spec/connection-manager.tex [hunk 3]
```diff
@@ -273,7 +356,212 @@ \section{Components}
...
+
+A key component of the design for decentralisation is the P2P governor. It
+is responsible for:
+\begin{itemize}
+\item managing the selection of upstream peers;
+\item managing the transitions of upstream peers between cold/warm/hot states;
+\item continuously making progress towards the target number of peers in each
+      state; and
+\item adjusting these targets over time to achieve a degree of `churn'.
+      \todo[inline]{need cross reference.}
```
Comment (bolt12) [inline docs/network-spec/connection-manager.tex:384]: This means we need a section on churn. Stealing from the most recent P2P blogpost might be a good idea

File: docs/network-spec/connection-manager.tex [hunk 4]
```diff
@@ -273,7 +356,212 @@ \section{Components}
...
+The inbound governor is responsible for starting, restarting and monitoring the
+server side of the mini-protocols.
+
+One of the high level design choices is that when a server side mini-protocol
+terminates cleanly (usually because the client chose to terminate it) then the
+server side of the mini-protocol should be restarted in its initial state in
+case the client wishes to use the protocol again later. It is the inbound
+governor that is responsible for doing this.
+
+\todo[inline]{this mux interaction lifetime may benefit from a seperate (sub-)section}
```
Comment (bolt12) [inline docs/network-spec/connection-manager.tex:467]: 👍🏼

File: docs/network-spec/connection-manager.tex [hunk 5]
```diff
@@ -273,7 +356,212 @@ \section{Components}
...
+      \hfil\textbf{Connection}\hfil\\
+      \hfil\textbf{Manager}\hfil
+    };
+
+  \draw[->] (p2p_governor)           -- (connection_manager);
+  \draw[->] (server.west)            -- (connection_manager.5);
+  \draw[<->] (connection_manager.355) -- (inbound_governor.west);
+\end{tikzpicture}
+\end{center}
+\end{designalternative}
```
Comment (bolt12) [inline docs/network-spec/connection-manager.tex:538]: Server does not interact with the inbound governor. Whenever the `Server` accepts a connection it calls `includeInboundConnection` which will notify the connection manager. Connection manager will perform handshake and inform Inbound Governor about the connection about it to start miniprotocols and will also inform the outbound governor about it if `Duplex` connection got negotiated in order for that inbound connection to be added to the `KnownPeer` set.

So in a way this design is already in place.

File: docs/network-spec/connection-manager.tex [hunk 6]
```diff
@@ -273,7 +356,212 @@ \section{Components}
...
+
+The inbound governor \todo[inline]{capture interfaces to other services}
+\todo[inline]{express intended sequencing more clearly}
```
Comment (bolt12) [inline docs/network-spec/connection-manager.tex:541]: What is this about?

File: docs/network-spec/mux.tex [hunk 7]
```diff
@@ -156,3 +171,5 @@ \section{Node-to-node and node-to-client protocol numbers}
   \caption{Node-to-client protocol numbers}
   \label{table:node-to-client-protocol-numbers}
 \end{table}
+
+\todo[inline]{Need subsection on how handshake is negotiated}
```
Comment (bolt12) [inline docs/network-spec/mux.tex:175]: What's the goal of this? To explain the protocol or the rules for the negotiation?

**Review by bolt12 — COMMENTED** [review]
This is an old draft PR and it was very hard to understand its scope. It has some outdated comments. It could be worth to redo it in a different PR and just pick up the useful bits

**Comment by bolt12** [comment]
cc @njd42 @coot what about this PR, should it be closed?

**Comment by coot** [comment]
It might be still relevant, we haven't changed anything fundamental in the connection manager since it was designed.

**Comment by coot** [comment]
Merged in #5001.

**Pull request closed without merging**